server/server.go

```diff
@@ -63,7 +70,18 @@ func New(cfg *Config) *Server {
 	r.HandleFunc("/download/{id:\\d+}/{name:.+}", srv.downloadHandler)
 	r.HandleFunc("/download/{id:\\d+}", srv.downloadHandler)
 	r.HandleFunc("/search/", srv.searchHandler)
-
+	apiRouter := r.PathPrefix("/api/").Subrouter()
+	key := os.Getenv("BOOKS_API_KEY")
+	if key == "" {
+		log.Printf("Warning: BOOKS_API_KEY not set; API disabled")
+	}
+	apiRouter.Use(func(next http.Handler) http.Handler {
+		return apiKeyMiddleware(key, next)
+	})
+	apiRouter.HandleFunc("/book/{id:\\d+}", srv.getBookHandler)
```

Comment:
I wonder if it's more idiomatic to use back ticks here, since we're using a regular expression, and the backslashes need not be escaped.